packages/studio-ui/src/AppWithLazyLoading.tsx

```diff
@@ -15,24 +15,26 @@ export default function AppWithLazyLoading() {
   ]);
   const componentsLoaded = loadedCount === totalCount;
   const [appLoaded, setAppLoaded] = useState(false);
+  const [stylesLoaded, setStylesLoaded] = useState(false);
+  const resourcesLoaded = appLoaded && stylesLoaded;
 
   useEffect(() => {
     loadComponents();
-    loadStyling();
+    void Promise.all(loadStyling()).then(() => setStylesLoaded(true));
     void AppPromise.then(() => setAppLoaded(true));
   }, []);
 
   return (
     <LoadingOverlay
-      loading={!componentsLoaded || !appLoaded}
+      loading={!(componentsLoaded && stylesLoaded && appLoaded)}
```

Comment:
nit: can we use `componentLoaded && resourcesLoaded` here?

Reply:
done!